presto-main/src/main/java/com/facebook/presto/sql/planner/PlanCanonicalInfoProvider.java

```diff
@@ -43,8 +43,9 @@ public interface PlanCanonicalInfoProvider
      * @param session Session for query being run
      * @param planNode Plan node to hash
      * @param strategy Strategy to canonicalize the plan node
+     * @param callMetaData Call metadata to get input table statistics when it's not available
```

Comment:
Add one more option. When it's true, if we didn't prefetch input statistics during query registration, we will return get it from metastore. When it's false, just return whatever in the cache.